pkg/render/monitor/monitor.go

```diff
@@ -58,6 +58,7 @@ const (
 	PrometheusDefaultPort           = 9090
 	PrometheusProxyPort             = 9095
 	PrometheusTLSSecretName         = "calico-node-prometheus-tls"
+	PrometheusClientTLSSecretName   = "calico-node-prometheus-client-tls"
```

Comment:
this seems to be specifically for calico-node, but is used for fluentd and other components as well that aren't calico/node -is this intended?

I think we should either have per-component TLS credentials, named as such, or a generic name that doesn't call out a specific component if it's shared between them.

Reply:
I added this to the issue description:
```
Some considerations that are made while working on this feature:
- An extra key-cert pair has been created for each client and/or server participating in the mTLS.
- Each server has to trust "calico-node-prometheus-client-tls", which is the secret that prometheus mounts. Our prometheus deployment unfortunately has the confusing name calico-node-prometheus, which at this point in time it not very specific any longer.
- The reason we do not re-use existing TLS secrets is because the certificate usages are not valid for the new mTLS scenario's. This would cause disruptions for all users upgrading to this version.
```

I am unsure if I really answer your question with this, but the reason for the name is because our prometheus has a poor name.

Reply:
Ya it's misleading but it's outside of the scope of this PR. We should have given prometheus a better name to start with.